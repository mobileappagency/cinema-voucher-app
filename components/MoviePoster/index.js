// @flow
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  Image,
  Text,
  TouchableOpacity,
  View
} from 'react-native'

type Movie = {
  title: string,
  genre: string,
  poster: string
}

type Props = {
  movie: Movie,
  onOpen: Function
}

export default class MoviePoster extends Component<Props> {
  static propTypes = {
    movie: PropTypes.object.isRequired,
    onOpen: PropTypes.func.isRequired
  }

  render () {
    const { movie, movie: { title, genre, poster }, onOpen } = this.props
    return (
      <TouchableOpacity onPress={() => onOpen(movie)}>
        <View >
          <Image source={{ uri: poster }} />
        </View>
        <Text numberOfLines={1}>{title}</Text>
        <Text numberOfLines={1}>{genre}</Text>
      </TouchableOpacity>
    )
  }
}
