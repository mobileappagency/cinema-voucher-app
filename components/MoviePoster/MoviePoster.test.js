import React from 'react'
import MoviePoster from './'
import { shallow } from 'enzyme'

describe('MoviePoster', () => {
  it('matches current snapshot', () => {
    const rendered = shallow(<MoviePoster movie={{}} onOpen={() => {}} />)

    expect(rendered).toMatchSnapshot()
  })
  
  it('executed onOpen', () => {
    const onOpenMock = jest.fn()

    const rendered = shallow(<MoviePoster movie={{}} onOpen={onOpenMock} />)

    rendered.simulate('press')

    expect(onOpenMock).toHaveBeenCalled()
  })
})
