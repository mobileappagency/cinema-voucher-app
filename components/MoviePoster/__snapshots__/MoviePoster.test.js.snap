// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster matches current snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <View>
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
  />
</TouchableOpacity>
`;
